### PR TITLE
CeloReserve.org V2

### PR DESCRIPTION
**Pull request by aaronmgdr**
# Main Features

* live info on reserve balances
* current stable token supply
* ratio of these
* fair market value of these 
* easy to add new stable tokens and reserve assets

# preview
https://stage-dot-celo-reserve-site-281118.oa.r.appspot.com/

Screen Shot 2021-03-07 at 7 36 42 PM
Screen Shot 2021-03-07 at 7 37 03 PM

#### A note on PR structure
IF you just want to review the data parts. please filter to the `src/providers/` and `src/services/` folders 

# auxilary features
* expanded jest testing with mocked responses for http reqs
* set proper cache headers 
* add detailed readme
* updated token icons to new version plus use svgs
* upgrade typescript to fix an issue with msw not supporting older version
* added babel plugin for emotion to get it to play nice with jest
* use semantically correct elements

# hightlights
* now loads in under a second

### CODE REVIEW

File: src/components/Ratios.tsx [hunk 1]
```diff
@@ -0,0 +1,68 @@
+import { css } from '@emotion/core'
+import useSWR from "swr"
+import Amount from 'src/components/Amount'
+import { BreakPoints } from 'src/components/styles'
+import { HoldingsApi } from "src/service/holdings"
+import StableValueTokensAPI from 'src/interfaces/stable-value-tokens'
+import { sumCeloTotal, sumNonCelo } from './Holdings'
+import { fetcher } from "src/utils/fetcher"
+
+function sumTotalHoldings(holdings:  HoldingsApi) {
```
Comment (aaronmgdr) [inline src/components/Ratios.tsx:10]: @martinvol this is where I sum everything to get ratios

File: src/components/ReserveAddresses.tsx [hunk 2]
```diff
@@ -39,28 +33,52 @@ function useCopy(hex) {
...
-function Address({ label, hex }) {
+const AddressDisplay = React.memo(function AddressDisplay({ label, hex, token }: DisplayProps) {
   const { onPress, justCopied } = useCopy(hex)
 
   return (
-    <div onClick={onPress} css={rootStyle}>
-      <h5 css={labelStyle}>{label}</h5>
-      <span css={css({ wordWrap: 'break-word' })}>
+    <div css={rootStyle}>
+      <h5 css={labelStyle}><a href={generatelink(token, hex)} target="_blank" rel="noopener">{label}</a></h5>
```
Comment (aaronmgdr) [inline src/components/ReserveAddresses.tsx:47]: link to relevant block explorer when clicking on label

File: src/components/ReserveAddresses.tsx [hunk 3]
```diff
@@ -39,28 +33,52 @@ function useCopy(hex) {
...
+      <span onClick={onPress} css={iconStyle}>
```
Comment (aaronmgdr) [inline src/components/ReserveAddresses.tsx:48]: copy address when clicking on itself

File: src/mocks/handlers.ts [hunk 4]
```diff
@@ -0,0 +1,212 @@
+import { rest } from 'msw'
+
+export const handlers = [
```
Comment (aaronmgdr) [inline src/mocks/handlers.ts:3]: This file is all the mock responses for third party APIs so its possible to write tests without depending on them being up / changing values

File: src/providers/BlockchainDotCom.test.ts [hunk 5]
```diff
@@ -0,0 +1,21 @@
+import { getBTCBalance, getBTCPrice } from "./BlockchainDotCom"
```
Comment (aaronmgdr) [inline src/providers/BlockchainDotCom.test.ts:1]: @martinvol 

Data gathering starts in this file down

File: src/service/stables.ts [hunk 6]
```diff
@@ -0,0 +1,47 @@
+import BigNumber from "bignumber.js"
```
Comment (aaronmgdr) [inline src/service/stables.ts:1]: @martinvol  files you are interested ends here

File: src/providers/BlockchainDotCom.test.ts (hunk 5 above)
Comment (martinvol) [reply src/providers/BlockchainDotCom.test.ts:1]: Do you have this functions mocked somewhere for tests? I can't see it

Comment (aaronmgdr) [reply src/providers/BlockchainDotCom.test.ts:1]: ah yes using` msw` package to mock the actual responses here src/mocks/handlers.ts

File: src/components/Button.tsx [hunk 7]
```diff
@@ -1,6 +1,6 @@
-/** @jsx jsx */
 
-import { css, jsx } from '@emotion/core'
+
+import { css, jsx } from '@emotion/react'
```
Comment (Henrynunez112) [inline src/components/Button.tsx:3]: Do you need the jsx here?

File: src/components/ChangeStory.tsx [hunk 8]
```diff
@@ -1,10 +1,10 @@
-/** @jsx jsx */
-import { css, jsx } from '@emotion/core'
+
+import { css, jsx } from '@emotion/react'
```
Comment (Henrynunez112) [inline src/components/ChangeStory.tsx:2]: Do you need the jsx here?

File: src/components/Footer.tsx [hunk 9]
```diff
@@ -1,5 +1,5 @@
-/** @jsx jsx */
-import { css, jsx } from '@emotion/core'
+
+import { css, jsx } from '@emotion/react'
```
Comment (Henrynunez112) [inline src/components/Footer.tsx:2]: Do you need to call jsx?

File: src/components/Markdown.tsx [hunk 10]
```diff
@@ -1,5 +1,5 @@
-/** @jsx jsx */
-import { css, jsx } from '@emotion/core'
+
+import { css, jsx } from '@emotion/react'
```
Comment (Henrynunez112) [inline src/components/Markdown.tsx:2]: jsx?

File: src/components/Section.tsx [hunk 11]
```diff
@@ -1,5 +1,5 @@
-/** @jsx jsx */
-import { css, jsx } from '@emotion/core'
+
+import { css, jsx } from '@emotion/react'
```
Comment (Henrynunez112) [inline src/components/Section.tsx:2]: jsx?

File: src/pages/legal/[doc].tsx [hunk 12]
```diff
@@ -1,5 +1,5 @@
-/** @jsx jsx */
-import { css, jsx } from '@emotion/core'
+
+import { css, jsx } from '@emotion/react'
```
Comment (Henrynunez112) [inline src/pages/legal/[doc].tsx:2]: jsx?

File: src/pages/404.tsx [hunk 13]
```diff
@@ -1,5 +1,5 @@
-/** @jsx jsx */
-import { css, jsx } from '@emotion/core'
+
+import { css, jsx } from '@emotion/react'
```
Comment (Henrynunez112) [inline src/pages/404.tsx:2]: jsx?

**Review by Henrynunez112 — COMMENTED** [review]
If we are using the latest emotion package we don't need to call jsx anymore. Other than that we're good.

**Pull request merged**